adapter/memquota2/memquota.go

```diff
@@ -48,11 +48,53 @@ type handler struct {
 	windows map[string]*rollingWindow
 
 	// the limits we know about
-	limits map[string]config.Params_Quota
+	limits map[string]*config.Params_Quota
+}
+
+// Limit is implemented by Quota and Override messages.
+type Limit interface {
+	GetMaxAmount() int64
+	GetValidDuration() time.Duration
+}
+
+// matchDimensions matches configured dimensions with dimensions of the instance.
+func matchDimensions(cfg map[string]string, inst map[string]interface{}) bool {
+	for k, val := range cfg {
+		rval := inst[k]
```

Comment:
so, this requires the dimensions to be in the same order as well, right?

Reply:
This is a map lookup - it shouldn't impose ordering constraints, right?

Reply:
Not really. we are checking *all* dimensions specified in adapterConfig. They can appear in any order.

Reply:
ah, ok. i was looking too quickly while trying to participate in that meeting.